assets/services/parachute/parachute.py

```diff
@@ -0,0 +1,209 @@
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_ACCEPTED
+            )
+            publish_parachute_status(mavlink_connection_out)
+        elif msg.param1 == mavutil.mavlink.MAVLINK_MSG_ID_COMPONENT_INFORMATION_BASIC:
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_ACCEPTED
+            )
+            publish_component_information_basic(mavlink_connection_out)
+        else:
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_DENIED
+            )
+
+    # MAV_CMD_SET_MESSAGE_INTERVAL
+    elif msg.command == mavutil.mavlink.MAV_CMD_SET_MESSAGE_INTERVAL:
+        if msg.param1 == mavutil.mavlink.MAVLINK_MSG_ID_PARACHUTE_STATUS:
+            if msg.param2 == -1:
+                parachute_status_thread_event.clear()
+            else:
+                parachute_status_thread_event.set()
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_ACCEPTED
+            )
+        else:
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_DENIED
+            )
+
+    # MAV_CMD_DO_PARACHUTE
+    elif msg.command == mavutil.mavlink.MAV_CMD_DO_PARACHUTE:
+        if msg.param1 == mavutil.mavlink.PARACHUTE_RELEASE:
+            if (
+                parachute_status[ParachuteStatusFields.ARM_STATUS.value]
+                & mavutil.mavlink.PARACHUTE_TRIGGER_FLAGS_FC
+            ):
+                if (
+                    parachute_status[ParachuteStatusFields.SAFETY_STATUS.value]
+                    & mavutil.mavlink.PARACHUTE_SAFETY_FLAGS_GROUND_CLEARED
+                ):
+                    mavlink_connection_out.mav.command_ack_send(
+                        msg.command, result=mavutil.mavlink.MAV_RESULT_ACCEPTED
+                    )
+                    update_parachute_status(
+                        ParachuteStatusFields.DEPLOYMENT_STATUS,
+                        int(mavutil.mavlink.PARACHUTE_DEPLOYMENT_TRIGGER_DRONE),
+                    )
+                else:
+                    mavlink_connection_out.mav.command_ack_send(
+                        msg.command, result=mavutil.mavlink.MAV_RESULT_TEMPORARILY_REJECTED
+                    )
+            else:
+                mavlink_connection_out.mav.command_ack_send(
+                    msg.command, result=mavutil.mavlink.MAV_RESULT_FAILED
+                )
+        else:
+            mavlink_connection_out.mav.command_ack_send(
+                msg.command, result=mavutil.mavlink.MAV_RESULT_DENIED
+            )
+
+    # MAV_CMD_SET_PARACHUTE_ARM
+    elif msg.command == mavutil.mavlink.MAV_CMD_SET_PARACHUTE_ARM:
+        arm_flags = int(msg.param1)
+        bitmask = int(msg.param2)
+        parachute_trigger_source_invalid = (
```

Comment:
trigger_source_invalid or trigger_flags_invalid?

Reply:
Also what does this variable represent? Have you manually checked the logic to see if the value is what you expect it to be? Isn't ENUM_END 256, which would make this value 509?

Reply:
renamed it to `valid_parachute_trigger_flags` to better represent what it is.

for mavlink enums, it seems to always be the largest value + 1. in the case of `PARACHUTE_TRIGGER_FLAGS_ENUM_END`, the largest value is the highest bit flag + 1, which gives 128 + 1